_search-plugins/sql/ppl/subsearch.md

```diff
@@ -0,0 +1,162 @@
+source = outer not exists [ source = inner | where a = c ] //search filtering with subquery
+source = table as t1 exists [ source = table as t2 | where t1.a = t2.a ] //table alias is useful in exists subquery
+source = outer | where exists [ source = inner1 | where a = c and exists [ source = nested | where c = e ] ] //nested
+source = outer | where exists [ source = inner1 | where a = c | where exists [ source = nested | where c = e ] ] //nested
+source = outer | where exists [ source = inner | where c > 10 ] //uncorrelated exists
+source = outer | where not exists [ source = inner | where c > 10 ] //uncorrelated exists
+source = outer | where exists [ source = inner ] | eval l = "nonEmpty" | fields l //special uncorrelated exists
+```
+
+## Scala subsearch
+
+### Syntax
+```sql
+where <field> = [ search source=... | ... | ... ]
+```
+
+### Usage
+```sql
+//Uncorrelated scalar subquery in Select
+source = outer | eval m = [ source = inner | stats max(c) ] | fields m, a
+source = outer | eval m = [ source = inner | stats max(c) ] + b | fields m, a
+
+//Uncorrelated scalar subquery in Where**
+source = outer | where a > [ source = inner | stats min(c) ] | fields a
+
+//Uncorrelated scalar subquery in Search filter
+source = outer a > [ source = inner | stats min(c) ] | fields a
+
+//Correlated scalar subquery in Select
+source = outer | eval m = [ source = inner | where outer.b = inner.d | stats max(c) ] | fields m, a
+source = outer | eval m = [ source = inner | where b = d | stats max(c) ] | fields m, a
+source = outer | eval m = [ source = inner | where outer.b > inner.d | stats max(c) ] | fields m, a
+
+//Correlated scalar subquery in Where
+source = outer | where a = [ source = inner | where outer.b = inner.d | stats max(c) ]
+source = outer | where a = [ source = inner | where b = d | stats max(c) ]
+source = outer | where [ source = inner | where outer.b = inner.d OR inner.d = 1 | stats count() ] > 0 | fields a
+
+//Correlated scalar subquery in Search filter
+source = outer a = [ source = inner | where b = d | stats max(c) ]
+source = outer [ source = inner | where outer.b = inner.d OR inner.d = 1 | stats count() ] > 0 | fields a
+
+//Nested scalar subquery
+source = outer | where a = [ source = inner | stats max(c) | sort c ] OR b = [ source = inner | where c = 1 | stats min(d) | sort d ]
+source = outer | where a = [ source = inner | where c =  [ source = nested | stats max(e) by f | sort f ] | stats max(d) by c | sort c | head 1 ]
+```
+
+## Relation subsearch
+
+### Syntax
+```sql
+join on <condition> [ search source=... | ... | ... ] [as alias]
+```
+
+### Usage
+```sql
+source = table1 | join left = l right = r on condition [ source = table2 | where d > 10 | head 5 ] //subquery in join right side
+source = [ source = table1 | join left = l right = r [ source = table2 | where d > 10 | head 5 ] | stats count(a) by b ] as outer | head 1
+```
+          
+# Examples
```

Comment:
Why we add TPC-H use case?  Can you consider use cases that are familiar to OpenSearch users?

Reply:
description updated.